src/garage/sampler/_functions.py

```diff
@@ -31,11 +32,11 @@ def _apply_env_update(old_env, env_update):
     if env_update is not None:
         if isinstance(env_update, EnvUpdate):
             return env_update(old_env), True
-        elif isinstance(env_update, gym.Env):
+        elif isinstance(env_update, (Environment, gym.Env)):
```

Comment:
```suggestion
        elif isinstance(env_update, Environment):
```

Reply:
gym.Env is actually needed here as test fixtures envs still inherit from gym.Env. See https://github.com/rlworkgroup/garage/blob/master/tests/garage/tf/samplers/test_tf_worker.py#L19

This can be changed once dummy envs are updated.